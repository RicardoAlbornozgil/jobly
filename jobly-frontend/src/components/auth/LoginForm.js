import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import Alert from "../common/Alert";
import "../../styles/Auth.css";

/** Login form.
 *
 * Shows form and manages updates to state on changes.
 * On submission:
 * - Calls login function prop
 * - Redirects to /companies route
 *
 * Routes -> LoginForm -> Alert
 * Routed as /login
 */

function LoginForm({ login }) {
  const navigate = useNavigate();
  const [formData, setFormData] = useState({
    username: "",
    password: "",
  });
  const [formErrors, setFormErrors] = useState([]);

  console.debug("LoginForm", "login=", typeof login, "formData=", formData, "formErrors=", formErrors);

  /** Handle form submit:
   *
   * Calls login func prop and, if successful, redirects to /companies.
   */

  async function handleSubmit(evt) {
    evt.preventDefault();
    const result = await login(formData);
    if (result.success) {
      navigate("/companies");
    } else {
      setFormErrors(result.errors);
    }
  }

  /** Update form data field */
  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  }

  return (
    <div className="main">
      <div className="LoginForm">
        <div className="form-card container col-md-6 col-lg-4">
          <h2 className="mb-3">Log In</h2>
          <div className="card">
            <div className="card-body">
              <form onSubmit={handleSubmit}>
                <div className="form-groups">
                  <div className="form-group">
                    <label htmlFor="username">Username</label>
                    <input
                      id="username"
                      name="username"
                      className="form-control"
                      value={formData.username}
                      onChange={handleChange}
                      autoComplete="username"
                      required
                    />
                  </div>
                  <div className="form-group">
                    <label htmlFor="password">Password</label>
                    <input
                      id="password"
                      type="password"
                      name="password"
                      className="form-control"
                      value={formData.password}
                      onChange={handleChange}
                      autoComplete="current-password"
                      required
                    />
                  </div>

                  {formErrors.length ? <Alert type="danger" messages={formErrors} /> : null}
                </div>
                <div className="card-footer">
                  <button className="btn btn-primary btn-block">
                    Submit
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default LoginForm;
