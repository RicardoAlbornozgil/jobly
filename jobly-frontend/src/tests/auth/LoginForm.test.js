import React from "react";
import { render } from "@testing-library/react";
import LoginForm from "../../components/auth/LoginForm";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <LoginForm />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
