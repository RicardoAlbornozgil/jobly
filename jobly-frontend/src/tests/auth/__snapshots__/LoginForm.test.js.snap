// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="LoginForm"
    >
      <div
        class="form-card container col-md-6 col-lg-4"
      >
        <h2
          class="mb-3"
        >
          Log In
        </h2>
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <form>
              <div
                class="form-groups"
              >
                <div
                  class="form-group"
                >
                  <label
                    for="username"
                  >
                    Username
                  </label>
                  <input
                    autocomplete="username"
                    class="form-control"
                    id="username"
                    name="username"
                    required=""
                    value=""
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    for="password"
                  >
                    Password
                  </label>
                  <input
                    autocomplete="current-password"
                    class="form-control"
                    id="password"
                    name="password"
                    required=""
                    type="password"
                    value=""
                  />
                </div>
              </div>
              <div
                class="card-footer"
              >
                <button
                  class="btn btn-primary btn-block"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
